### PR TITLE
140 rework filter repr

### PR DESCRIPTION
**Pull request by sherzinger**
closes #140

@peterbanda can you please have a look if the UI behaves as it should?
To me it looks fine, but you know the front-end better than I do.

### CODE REVIEW

File: app/views/filter/conditionPanel.scala.html [hunk 1]
```diff
@@ -1,57 +1,28 @@
...
-        }
-
-    valueOrLabel match {
-        case None => "undefined"
-        case Some(s) if s.isEmpty => "undefined"
-        case Some(s) => shorten(s, 80)
-    }
-}
-
-@defining((filter.isEmpty || filter.get._id.isEmpty) && filter.map(_.conditions.size).getOrElse(0) < 4) { showPanel =>
```
Comment (peterbanda) [inline app/views/filter/conditionPanel.scala.html]: So a full filter is always displayed regardless of the number of conditions... I think it's good but might be revisited if the filter area gets too big.

File: app/views/filters/displayConditions.scala.html [hunk 2]
```diff
@@ -1,22 +1,11 @@
 @import org.incal.core.FilterCondition
+@import org.ada.web.util.{conditionTypeToReadableOperator, valueOrUndefined}
 
 @(conditions: Traversable[FilterCondition])
 
-@valueOrUndefined(condition: FilterCondition) = @{
```
Comment (peterbanda) [inline app/views/filters/displayConditions.scala.html]: Good catch! Didn't realize `valueOrUndefined` was at two places. Glad you put it into utils.

File: public/stylesheets/main.css [hunk 3]
```diff
@@ -392,9 +392,46 @@ a.carousel-control.right{
     }
 }
 
+.condition-modal-form-row {
```
Comment (peterbanda) [inline public/stylesheets/main.css]: Indeed, flex mode makes it all nicely aligned (horizontally).

File: app/views/filter/conditionPanel.scala.html (hunk 1 above)
Comment (sherzinger) [reply app/views/filter/conditionPanel.scala.html]: Indeed, for long names we need to trim. I will do this as part of ada-discovery/ada-issues#144

**Pull request merged**